lib/promscrape/scrapework.go

```diff
@@ -510,9 +510,13 @@ func (sw *scrapeWork) processScrapedData(scrapeTimestamp, realTimestamp int64, b
 	}
 	// body must be released only after wc is released, since wc refers to body.
 	if !areIdenticalSeries {
-		// Send stale markers for disappeared metrics with the real scrape timestamp
-		// in order to guarantee that query doesn't return data after this time for the disappeared metrics.
-		sw.sendStaleSeries(lastScrape, bodyString, realTimestamp, false)
+		// disable sendStaleSeries, since it is impossible to track which series became stale
+		// when seriesLimiter discards extra series.
+		if !sw.seriesLimitExceeded {
```

Comment:
It is better to apply the series limit to stale markers inside `sendStaleSeries()` function before sending them to the remote storage.